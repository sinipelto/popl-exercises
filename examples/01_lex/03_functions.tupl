define Squares[arg]
begin
  [arg * arg].
end.

[1..10] | each: Squares -> Print.
