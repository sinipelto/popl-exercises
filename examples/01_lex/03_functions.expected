LexToken(DEFINE,'define',1,0)
LexToken(funcIDENT,'Squares',1,7)
LexToken(LSQUARE,'[',1,14)
LexToken(varIDENT,'arg',1,15)
LexToken(RSQUARE,']',1,18)
LexToken(BEGIN,'begin',2,20)
LexToken(LSQUARE,'[',3,28)
LexToken(varIDENT,'arg',3,29)
LexToken(MULT,'*',3,33)
LexToken(varIDENT,'arg',3,35)
LexToken(RSQUARE,']',3,38)
LexToken(DOT,'.',3,39)
LexToken(END,'end',4,41)
LexToken(DOT,'.',4,44)
LexToken(LSQUARE,'[',6,47)
LexToken(NUMBER_LITERAL,1,6,48)
LexToken(DOUBLEDOT,'..',6,49)
LexToken(NUMBER_LITERAL,10,6,51)
LexToken(RSQUARE,']',6,53)
LexToken(PIPE,'|',6,55)
LexToken(EACH,'each',6,57)
LexToken(COLON,':',6,61)
LexToken(funcIDENT,'Squares',6,63)
LexToken(RARROW,'->',6,71)
LexToken(funcIDENT,'Print',6,74)
LexToken(DOT,'.',6,79)
